doc/Programming-guide/Your-first-Kaa-application/index.md

```diff
@@ -399,6 +416,146 @@ int main(void)
 
 <div id="app-obj-c" class="tab-pane fade" markdown="1" >
 
+```objc
+#import "ViewController.h"
+
+@import Kaa;
+
+@interface ViewController () <KaaClientStateDelegate, ConfigurationDelegate, ProfileContainer>
+
+@property (nonatomic, strong) id<KaaClient> kaaClient;
+@property (nonatomic, weak) NSTimer *samplingTimer;
+
+@end
+
+static const NSInteger defaultSamplePeriod = 1;
+
+@implementation ViewController
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+
+    NSLog(@"Your first application started");
+
+    // Create a Kaa client and add a listener which displays the Kaa client status
+    // as soon as the Kaa client is started.
+    self.kaaClient = [KaaClientFactory clientWithStateDelegate:self];
+    
+    // Configure log uploading strategy so that logs are uploaded upon adding a new log. Set maximum parallel log uploads to 1.
+    RecordCountLogUploadStrategy *logUploadStrategy = [[RecordCountLogUploadStrategy alloc] initWithCountThreshold:1];
+    logUploadStrategy.maxParallelUploads = 1;
+    [self.kaaClient setLogUploadStrategy:logUploadStrategy];
+    
+    
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+    NSString *configurationPath = [[paths objectAtIndex:0] stringByAppendingPathComponent:@"savedconfig.cfg"];
+
+    // Persist configuration in a local storage to avoid downloading it each time the Kaa client is started.
+    [self.kaaClient setConfigurationStorage:[SimpleConfigurationStorage storageWithPath:configurationPath]];
+    
+    // Add a delegate which displays the Kaa client configuration each time it is updated.
+    [self.kaaClient addConfigurationDelegate:self];
+    
+    // Add a delegate which returns Kaa profile upon request.
+    [self.kaaClient setProfileContainer:self];
+    
+    // Start the Kaa client and connect it to the Kaa server.
+    [self.kaaClient start];
+}
+
+#pragma mark - Supporting methods
+
+/**
+ * Schedules new timer each time when sample period was updated.
+ */
+
+- (void)createSamplingTimerWithInterval:(NSInteger)period {
+    if (self.samplingTimer) {
+        [self.samplingTimer invalidate];
+    }
+    if (period >= 0) {
```

Comment:
Please, replace this `if` block to the next one:
```
if (period == 0) {
    self.samplingTimer = [NSTimer scheduledTimerWithTimeInterval:defaultSamplePeriod target:self selector:@selector(generateTemperature) userInfo:nil repeats:YES];
} else if (period >= 0) {
    self.samplingTimer = [NSTimer scheduledTimerWithTimeInterval:period target:self selector:@selector(generateTemperature) userInfo:nil repeats:YES];
}
```

In your variant if the **period** variable is zero, timer value will not be equal to zero (NSTimer initialization duplicated).